### PR TITLE
plan: remove unionscan schema.

### PR DESCRIPTION
**Pull request by hanfei1991**
add logicalUnionScan

### CODE REVIEW

File: plan/logical_plan_builder.go [hunk 1]
```diff
@@ -1583,12 +1575,12 @@ func (b *planBuilder) buildDataSource(tn *ast.TableName) LogicalPlan {
 		columns = tbl.Cols()
 	}
 	var pkCol *expression.Column
```
Comment (winoros) [inline plan/logical_plan_builder.go]: so `handleCol` is better.

File: plan/column_pruning.go [hunk 2]
```diff
@@ -157,16 +157,21 @@ func (p *Union) PruneColumns(parentUsedCols []*expression.Column) {
 }
 
 // PruneColumns implements LogicalPlan interface.
+func (p *LogicalUnionScan) PruneColumns(parentUsedCols []*expression.Column) {
+	for _, col := range p.schema.TblID2Handle {
+		parentUsedCols = append(parentUsedCols, col[0])
+	}
+	p.children[0].(LogicalPlan).PruneColumns(parentUsedCols)
+	p.SetSchema(p.children[0].Schema())
```
Comment (XuHuaiyu) [inline plan/column_pruning.go]: if `PkIsHandle` is false, `p.Schema` can remove the last col of p.children[0].Schema() ?

File: plan/physical_plan_builder.go [hunk 3]
```diff
@@ -69,6 +69,19 @@ func (p *requiredProp) enforceProperty(task task, ctx context.Context) task {
 	return sort.attach2Task(task)
 }
 
+func (p *PhysicalUnionScan) getChildrenPossibleProps(prop *requiredProp) [][]*requiredProp {
+	if !prop.isEmpty() {
```
Comment (XuHuaiyu) [inline plan/physical_plan_builder.go]: Sort can be pushed down to UnionScan.
refer UnionScanExec.compare

File: plan/column_pruning.go (hunk 2 above)
Comment (hanfei1991) [reply plan/column_pruning.go]: the following project can prove there is no extra column.

File: plan/physical_plan_builder.go (hunk 3 above)
Comment (hanfei1991) [reply plan/physical_plan_builder.go]: ok

File: plan/column_pruning.go (hunk 2 above)
Comment (XuHuaiyu) [reply plan/column_pruning.go]: If set a proper schema for UnionScan,
the Prj may be eliminated.

Comment (hanfei1991) [reply plan/column_pruning.go]: The length of union scan’ schema is equal to the length of row it outputs. The projection can’t be eliminate.

**Pull request merged**